src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        if (size == elements.length) {
+            resize();
+        }
+        elements[size++] = value;
+    }
 
+    private void resize() {
+        int newCapacity = (int) (elements.length * 1.5);
+        T[] newArray = (T[]) new Object[newCapacity];
+        for (int i = 0; i < size; i++) {
+            newArray[i] = elements[i];
+        }
+        elements = newArray;
     }
```

Comment:
Use System.arraycopy() to move your array elements when resizing instead of manually copying them in a loop.